app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,37 @@
+class ApplicationsController < ApplicationController 
+
+  def show 
+    @application = Application.find(params[:id])
+    if params[:search].present?
+      @pets = Pet.search(params[:search])
+    end
+  end
+
+  def create 
+    application =Application.new(application_params)
+    if application.save 
+      redirect_to "/applications/#{application.id}"
+    else
+      redirect_to"/applications/new"
+      flash[:alert] = "Error: All fields must be filled out"
+    end
+  end
```

Comment:
Great job using flash messaging and sad path conditional logic for when things go correctly, and when they don't.